smart-contracts/wasm-chain-integration/src/utils.rs

```diff
@@ -175,66 +208,422 @@ impl<'a, R: RngCore, BackingStore: trie::BackingStoreLoad> machine::Host<Artifac
+
+                    stack.push_value(bytes_written)
+                } else {
+                    stack.push_value(-1i32)
                 }
             }
-        } else if f.matches("concordium", "debug_print") {
-            let (filename, line, column, msg) = extract_debug(memory, stack)?;
-            self.debug_events.push(EmittedDebugStatement {
-                filename,
-                line,
-                column,
-                msg,
-                remaining_energy: 0.into(), // debug host does not have energy.
-            });
-        } else if f.matches("concordium", "state_lookup_entry") {
-            host::state_lookup_entry(memory, stack, &mut energy, &mut self.state)?;
-        } else if f.matches("concordium", "state_create_entry") {
-            host::state_create_entry(memory, stack, &mut energy, &mut self.state)?;
-        } else if f.matches("concordium", "state_delete_entry") {
-            host::state_delete_entry(memory, stack, &mut energy, &mut self.state)?;
-        } else if f.matches("concordium", "state_delete_prefix") {
-            host::state_delete_prefix(memory, stack, &mut energy, &mut self.state)?;
-        } else if f.matches("concordium", "state_iterate_prefix") {
-            host::state_iterator(memory, stack, &mut energy, &mut self.state)?;
-        } else if f.matches("concordium", "state_iterator_next") {
-            host::state_iterator_next(stack, &mut energy, &mut self.state)?;
-        } else if f.matches("concordium", "state_iterator_delete") {
-            host::state_iterator_delete(stack, &mut energy, &mut self.state)?;
-        } else if f.matches("concordium", "state_iterator_key_size") {
-            host::state_iterator_key_size(stack, &mut energy, &mut self.state)?;
-        } else if f.matches("concordium", "state_iterator_key_read") {
-            host::state_iterator_key_read(memory, stack, &mut energy, &mut self.state)?;
-        } else if f.matches("concordium", "state_entry_read") {
-            host::state_entry_read(memory, stack, &mut energy, &mut self.state)?;
-        } else if f.matches("concordium", "state_entry_write") {
-            host::state_entry_write(memory, stack, &mut energy, &mut self.state)?;
-        } else if f.matches("concordium", "state_entry_size") {
-            host::state_entry_size(stack, &mut energy, &mut self.state)?;
-        } else if f.matches("concordium", "state_entry_resize") {
-            host::state_entry_resize(stack, &mut energy, &mut self.state)?;
-        } else {
-            bail!("Unsupported host function call.")
+            "log_event" => {
+                let event_length = unsafe { stack.pop_u32() };
+                let event_start = unsafe { stack.pop_u32() };
+
+                // TODO: Log can be full and messages can be too long, but it is unspecified
+                // what the limits are. Find out, document and fail if either is too long.
```

Comment:
[The relevant limits can be found here](https://github.com/Concordium/concordium-base/blob/main/smart-contracts/wasm-chain-integration/src/constants.rs#L12)